src/TropicalGeometry/variety_binomial.jl

```diff
@@ -0,0 +1,41 @@
+################################################################################
+#
+#  Tropicalization of binomial ideals
+#
+#  WARNING: assumes without test that `gens(I)` is binomial
+#
+################################################################################
+
+function tropical_variety_binomial(I::MPolyIdeal,nu::TropicalSemiringMap; weighted_polyhedral_complex_only::Bool=false)
+    ###
+    # Construct matrix of exponent vector differences
+    # and vector of coefficient valuation differences
+    ###
+    G = gens(I)
+    A = matrix(ZZ,[first(expv)-last(expv) for expv in exponents.(G)])
+    b = [ QQ(nu(last(coeff)/first(coeff))) for coeff in coefficients.(G)]
```

Comment:
Unfortunately it turns out my suggestion was wrong, `last` is not supported by these iterators. As a quick fix (with a result that still is IMHO a bit more legible than the original code), one can do this:
```suggestion
    A = matrix(ZZ,[first(expv)-last(expv) for expv in collect.(exponents.(G))])
    b = [ QQ(nu(last(coeff)/first(coeff))) for coeff in collect.(coefficients.(G))]
```
Luckily as long as the polynomials are binomials it shouldn't matter much.

Reply:
I am going to apply this now to hopefully unbreak the tests.